### DISCUSSION TITLE
Printer config SELPHY CP

### DISCUSSION THREAD

**Discussion by Striker62** [photobooth-app/photobooth-app#293] [Howtos and camera specific topics]
Hi,
I add a CP1500 to my box.
As far as I understood is ther no need for detailed parameter in the command line.

lp -d Canon_SELPHY_CP1500 {filename}

did not print . in the log i found:

<html><body>

ERROR | 11.Jun.24 12:40:16 | RequestValidationError:  [{'type': 'missing', 'loc': ('body', 'printer', 'print', 1, 'trigger'),  'msg': 'Field required', 'input': {'name': 'SELPHY', 'processing':  {'printing_command': 'lp -d Canon_SELPHY_CP1500 {filename}',  'printing_blocked_time': 10}}}] | photobooth.application | validation_exception_handler | 85
-- | -- | -- | -- | -- | --

</body>
</html>

Did I miss some thing? how das your Commandline looks like for the cp1300

**Comment by mgineer85** [comment]
You just need to set any trigger in the config.

**Comment by Striker62** [comment]
Ah ok I see. Thanks

the printer now react but I get the message 

Inkompatible bilder nicht druckbar ?

Any Idea ?

**Reply by mgineer85** [reply]
Please search old discussions and general google for printing on Linux. I think you might need the latest Gutenberg printerdriver compiled on your system.

**Reply by rafffx** [reply]
hello, two think , not sure if it help, but try to be sure the paper tank is well install sometime if not it's can't warning that wrong paper is instal, then for this issue, maybe try to match with the best resolution the cam you use, to fit the ratio of canon selphy  100X148, ratio 1.48   ,   maybe the cam you use create picture of 1.72 ratio for exemple and the canon printer don't like that even if there is an option to fit to scale but , i hope it could help

**Comment by rafffx** [comment]
Hello, i got another issue with canon selphy, cp 1500, everything work fine , until i get my first print. then i get may printer state stuck at page 1, 100% , adn not refreshing after a single still picture print.

did someone know how to fix this problem, maybe need to do something to make the pirnter understand that there only one page and not foloowing, of just a way to refresh printer state after page one is done ???  how do i forget an config step on gutenprint or cups config ?
thanks a lot

**Reply by mgineer85** [reply]
Hey, I unfortunately do not have a solution for you.
What it comes down to is that you need to find a command line printing fine a JPG. When this works, then it works in the photobooth-app because it only invokes a command :)

**Reply by Striker62** [reply]
Hello raffix,

do you have a printer manger software installed.
If you use an raspberry pi you can use cups.
Here you can check the status of the printqueue and delete old prints.
I had the similar issues at the beginning. My printer goes some times from activ to inactive. So I had to activate it again.
Then all queued print started ;-) ( better check the queue before you activate )
I tried a lot with the testpage printing of cups, that helped me .

**Reply by rafffx** [reply]
@Striker62, I have the cups installation for the cp1500, and I've already checked the queue and cleaned it up before testing it, but every time, after the first print that goes correctly, it crashes after the first one, and can't print another one after that, without manually cleaning the queue to get it working again, 
I just want to avoid this process of manually cleaning the queue. I'd like just one print to be 100% completed statut , then it can print again afterwards.

**Reply by PixelAlarm** [reply]
Did you try adding this to your command?
cupsenable printer_name;

This might get the Printer out of the stuck state.

**Reply by rafffx** [reply]
> Did you try adding this to your command?
> 
> cupsenable printer_name;
> 
> 
> 
> This might get the Printer out of the stuck state.

I will give it a try , thanks a lot.

**Reply by rafffx** [reply]
> Did you try adding this to your command? cupsenable printer_name;
> 
> This might get the Printer out of the stuck state.

it does help on my issue after trying :(. thanks a lot anyway :)

**Comment by Striker62** [comment]
find here my print command:

lp -d SELPHY-CP1500 -o landscape -o fit-to-page {filename}

The printer name can be another on your Pi , I changed it on my pi

**Reply by rafffx** [reply]
i used the same  , but after the first print  from a clean queue, it stuck the queue with the statut pages 1, 100%, and doesn't want to print because of that, without manually clean it.

**Comment by Striker62** [comment]
did you try the test page?

**Reply by rafffx** [reply]
yes, its works for the test page, but need to check the statut after that> let me try it again then check the statut this time

**Reply by Striker62** [reply]
i can try tomorrow to export my cups printer definition.

**Reply by rafffx** [reply]
just for the explanation , i tried  print test, i can handle two print in row. it working, but nothing appear on job.

but when i tried with photobooth-app i get this issue of sticking state of page 1 100% for the second print then pending for the third, but even if i cancelled some of the previous to test again, with will remain stuck, until i reboot the printer : (See picture)

![issue_printer](https://github.com/user-attachments/assets/4996bf31-af98-49c7-979c-656d2429da58)


i don't understand the new state "printing page 1, 100 %" stuck

**Reply by Striker62** [reply]
I nerver had that issue . 
It tooks some try and error with the driver but did not compile one . Not sure if I'm choosed the CP1300 . 
I hope the cups file can help you.
 
[cups-raspi-photobooth-mini.tar.gz](https://github.com/user-attachments/files/17130914/cups-raspi-photobooth-mini.tar.gz)

**Reply by Striker62** [reply]
by the way I use the cups gui not the web page.

**Reply by Striker62** [reply]
For your Info!!! I had to change the user of the files to export them. so do not copy my files to the /etc/cups dir.
Pls check the permissons of the original files in your system

**Reply by rafffx** [reply]
tryiing everything , and trying to check what happenm from gui for now, i see that : first it print right, but nothing appear on  queue, and printer is available , and showing in print list to be add .   i got my printer print as expecting anyway   .      image 01

at the second attemp just after the first print, I got a state appear on queue and in t the printer state about the print page 1, 100% and also see that the printer disappear from list of printer, like it is not recocgnize anymore after the first print .    image 02



![pb02](https://github.com/user-attachments/assets/308e83a8-fe47-4fe3-a778-3b6bba857330)
![pb03](https://github.com/user-attachments/assets/ec52d395-a7fa-4ef1-8e7c-7854d35a3f42)

**Reply by Striker62** [reply]
Hi, I see you have the printer via USB connected. I use it as a Networkprinter.
I did not try USB because I want to use the QR-code sharing so I setup an accesspoint and use it for the printer also.

**Reply by Striker62** [reply]
![raspi-photobooth-mini - Remotedesktopverbindung 30 09 2024 09_10_51](https://github.com/user-attachments/assets/f2c8a8b0-1b3c-4d01-a204-cd5921ce7f35)

**Reply by mgineer85** [reply]
Unfortunately I cannot guide you to solve the issue finally.
Maybe it's worth to test also via WiFi? I heard these Selphys could have issues with USB/Linux. Could be worth a try.

**Reply by rafffx** [reply]
thanks a lot for all your help everyone, really appreciate.
I will give it a try,  with the wifi.
cheers !:)
